CHANGELOG.md

```diff
@@ -1,3 +1,87 @@
+### citus v10.2.0 (September 11, 2021) ###
+
+* Adds PostgreSQL 14 support
+
+* Adds hash & btree index support for columnar tables
+
+* Adds helper UDFs for easy time partition management:
+  `get_missing_time_partition_ranges`,
+  `create_time_partitions`,
+  `drop_old_time_partitions`
+
+* Adds a parameter to cleanup metadata when metadata syncing is stopped
+
+* Adds propagation of ALTER SEQUENCE
+
+* Adds support for ALTER INDEX ATTACH PARTITION
+
+* Adds support for CREATE INDEX ON ONLY
+
+* Allows more graceful failovers when replication factor > 1
+
+* Enables chunk group filtering to work with Params for columnar tables
+
+* Enables qual push down for joins including columnar tables
+
+* Enables transferring of data using binary encoding by default on PG14
+
+* Fixes a bug that allowed moving of shards belonging to a reference table
+
+* Fixes a bug about int and smallint sequences on MX
+
+* Fixes a bug that cause partitions to have wrong distribution key after
+  DROP COLUMN
+
+* Fixes a bug that caused `worker_append_table_to_shard` to write as superuser
```

Comment:
We also backported this: https://github.com/citusdata/citus/pull/5256  but never released earlier versions.

What is our policy on such cases. We just keep the changelog entry here?

Reply:
I do not have enough context on this PR.

My personal opinion on what to include on `10.2.0` is the following: Add all entries that are introduced after `10.1.0` including those that are already listed under `10.1.1` and `10.1.2`. This way the user can only take a look at the major versions to understand about the capability of each version.

For example, the end user may not understand that all the fixes in `10.1.1` are also shipped in `10.2.0`. We as Citus developers know about this pattern, but I am not so sure if the end users feel the same way.

Reply:
If we didn't release yet, then I think we should have changelog entry here.
Otherwise, we don't put it to the new release's changelog entries.

Reply:
If a bug never existed on any earlier version, then it makes more sense to drop the entry for the bugfix. Though I am not so sure if that's the case for this PR.

It would be nice if we can have `cherry-pick-X.Y` on the PR to understand which versions are actually affected.

Reply:
This is back ported up to 9.4 (including enterprise): https://github.com/citusdata/citus/commits/release-9.4

I guess our typical approach is to have this entry on 9.4.X change logs? I think to be consistent, I have a slight preference to have it only on 9.4.X

Reply:
> I guess our typical approach is to have this entry on 9.4.X change logs?

We usually add the item to all the versions that we backported the fix to.

For example, a user that uses 10.0 will be able to see if a patch release can solve an issue that they run into.

For example the following entry is in `9.4.5`, `9.5.5`, `10.0.4` and `10.1.0` and it makes sense to me.
> Fixes an issue that could cause citus_finish_pg_upgrade to fail

Reply:
Alright, then lets keep it here. Thanks for the explanation